### PR TITLE
Feature/localisation

### LINKED ISSUE #3: Localization (3 languages)   i18next
Add react-i18next localisation
Decide whether it's worth to use gatsby default way

### PR DESCRIPTION
**Pull request by vadzim-sasonkin**
You can find localization messages in src/data/messages in JSON format. 
Please look at the header component as a usage example.

More at https://react.i18next.com/

P.S. @pnhepw  I have utilized Local storage instead of url paths as it comes with react-intl18next package out of the box, but we can discuss it still.

### CODE REVIEW

**Comment by rss-maksim** [comment]
@OrionT I think it's not very important now.
But it seems to me it's better to use url because in such way user will be able to share site in proper language.

File: src/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+import "../localisation"
```
Comment (rss-maksim) [inline src/components/app.jsx]: We can agree separate by empty line imports from node_modules and from local modules

File: src/components/LocaleSelector/LocaleSelector.jsx [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-import React from 'react'
-import { Link } from '@material-ui/core';
+import React from "react"
+import { Link } from "@material-ui/core"
 
-const LocaleSelector = ({ locales }) => {
+const LocaleSelector = ({ locales, onSwitch }) => {
   return locales.map(({ icon: Icon, name, url }) => (
-    <Link href={url} key={name} title={name.toUpperCase()}>
+    <Link key={name} title={name.toUpperCase()} onClick={() => onSwitch(name)}>
```
Comment (rss-maksim) [inline src/components/LocaleSelector/LocaleSelector.jsx]: It's better to pass handler to component as ```() => onSwitch(name)``` if you can. In such way link on function doesn't change

File: src/components/app.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+import Layout from "./layout"
+
+const App = ({ children }) => {
```
Comment (rss-maksim) [inline src/components/app.jsx]: You can simplify:
```
const App = ({ children }) => <Layout>{children}</Layout>
```

File: src/components/header.jsx [hunk 4]
```diff
@@ -1,52 +1,63 @@
...
-      </Typography>
-      <Box className="header-locale-selector">
-        <LocaleSelector locales={locales} />
-      </Box>
-    </Toolbar>
-  </AppBar>
-)
+const Header = ({ siteTitle }) => {
+  const { t, i18n } = useTranslation()
+  const handleChangeLanguage = lang => {
```
Comment (rss-maksim) [inline src/components/header.jsx]: You can simplify:
```
const handleChangeLanguage = lang => i18n.changeLanguage(lang)
```
Or even 
```
const handleChangeLanguage = lang => () => i18n.changeLanguage(lang)
```
in order not to use arrow function in component LocaleSelector

File: src/pages/index.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
-import React from "react"
+import React, { Suspense } from "react"
```
Comment (rss-maksim) [inline src/pages/index.js]: Do you plan use Suspense in this file in the future?

**Comment by vadzim-sasonkin** [comment]
> @OrionT I think it's not very important now.
> But it seems to me it's better to use url because in such way user will be able to share site in proper language.

@rss-maksim Ok, let's discuss it on a meeting and decide whether a new task is needed.

File: src/components/app.jsx (hunk 1 above)
Comment (vadzim-sasonkin) [reply src/components/app.jsx]: +

File: src/components/app.jsx (hunk 3 above)
Comment (vadzim-sasonkin) [reply src/components/app.jsx]: +

File: src/pages/index.js (hunk 5 above)
Comment (vadzim-sasonkin) [reply src/pages/index.js]: No, it happened occasionally. We can think over adding a rule to eslint.

**Review by rss-maksim — APPROVED** [review]

File: src/components/header.jsx (hunk 4 above)
Comment (vadzim-sasonkin) [reply src/components/header.jsx]: done

File: src/components/LocaleSelector/LocaleSelector.jsx (hunk 2 above)
Comment (vadzim-sasonkin) [reply src/components/LocaleSelector/LocaleSelector.jsx]: done

**Pull request merged**